javatests/arcs/core/analysis/testdata/fail_field_claim_propagates_type_variables.arcs

```diff
@@ -0,0 +1,27 @@
+// #Ingress: P1
+// #Fail: hc:P4.foo.b is untrusted
+particle P1
+  foo: writes Foo {a: Text, b: Number}
+  claim foo.a is trusted
+  claim foo.b is untrusted
+particle P2
+  foo: reads ~a with {a: Text}  // Constraints
+  bar: writes ~b
```

Comment:
This is... weird. Or wrong?

We apparently never check for it, but a particle would not be able to write ~a without ever reading ~a. How can it know what ~a is?

I guess with all the reflective capabilities we're adding, it could... so maybe it's not so crazy after all, but still worth flagging out that this is unexpected.

Reply:
Let me suggest a more typical example:

```
particle P1
  foo: writes Foo {a: Text, b: Number}
  claim foo.a is trusted
  claim foo.b is untrusted
particle P2
  foo: reads ~a with {a: Text}
  bar: writes Foo {a: Text, b: Number}
particle P3
  foo: reads Foo {b: Number}
  check foo.b is untrusted
recipe R
  P1
    foo: writes h1
  P2
    foo: reads h1
    bar: writes h2
  P3
    foo: reads h2
```

This one is interesting, because if you connect P1 to P3 it would be fine, but P2 is routing the data from Foo.b into Foo.{a, b}, which then should cause the check to fail.

Reply:
I expected  `~b` to get resolved because it gets hooked up with `P3.baz`. However, you are right this may fail code gen. Where do you want me to flag it? Create a bug?